### PR TITLE
adding snaspshot mount testcases

### PR DESCRIPTION
**Pull request by maheshbhoge**
@savitha-p , @rajneeshkapoor , @sachin-trilio Please review

### CODE REVIEW

File: tempest/api/workloadmgr/snapshot/test_snapshot_mount.py [hunk 1]
```diff
@@ -93,3 +93,137 @@ def test_1_filesearch_default_parameters(self):
...
             reporting.set_test_script_status(tvaultconf.FAIL)
             reporting.test_case_to_write()
+
+
+    @test.attr(type='smoke')
+    @test.idempotent_id('28fd0710-ef00-42b0-93f0-9dbb3ffc5bee')
+    def test_2_umount_snapshot(self):
+        reporting.add_test_script(str(__name__) + "_umount_snapshot")
+        try:
+            if self.exception != "":
```
Comment (kiranpawar7) [inline tempest/api/workloadmgr/snapshot/test_snapshot_mount.py]: Do we require this if condition?? As this test case is not followed by any prerequisite call.

File: tempest/api/workloadmgr/snapshot/test_snapshot_mount.py [hunk 2]
```diff
@@ -93,3 +93,137 @@ def test_1_filesearch_default_parameters(self):
...
+            instances_ids = self.instances_ids
+            snapshot_ids = self.snapshot_ids
+            wid = self.wid
+            volumes_ids = self.volumes_ids
+            security_group_id = self.security_group_id
+            fvm_id = self.fvm_id
+            dismount_snapshot_id = snapshot_ids[0]
+            floating_ips_list = self.floating_ips_list
+
+            LOG.debug("dismount snasphot of a  snapshot")
```
Comment (kiranpawar7) [inline tempest/api/workloadmgr/snapshot/test_snapshot_mount.py]: replace dismount as umount and can be given as "Umount snapshot"

File: tempest/api/workloadmgr/snapshot/test_snapshot_mount.py [hunk 3]
```diff
@@ -93,3 +93,137 @@ def test_1_filesearch_default_parameters(self):
...
+            output_list = self.validate_snapshot_mount(self.ssh)
+            self.ssh.close()
+            
+            for i in output_list:
+                if '/home/ubuntu/mount_data_b' not in i:
+                   LOG.debug("connect to fvm and check mountpoint is unmounted from FVM instance")
+                   reporting.add_test_step("Verify that mountpoint is unmounted from FVM instance", tvaultconf.PASS)
+                   reporting.test_case_to_write()
+                else:
+                   LOG.debug("mount snapshot with full snapshot is unsuccessful on FVM")
```
Comment (kiranpawar7) [inline tempest/api/workloadmgr/snapshot/test_snapshot_mount.py]: Replace mount to umount

File: tempest/api/workloadmgr/snapshot/test_snapshot_mount.py [hunk 4]
```diff
@@ -93,3 +93,137 @@ def test_1_filesearch_default_parameters(self):
...
+                   reporting.add_test_step("Verify that mountpoint is unmounted from FVM instance", tvaultconf.FAIL)
+                   raise Exception ("mountpoint is not showing on FVM instance")
```
Comment (kiranpawar7) [inline tempest/api/workloadmgr/snapshot/test_snapshot_mount.py]: Exception describing  wrong message.

File: tempest/api/workloadmgr/snapshot/test_snapshot_mount.py [hunk 5]
```diff
@@ -93,3 +93,137 @@ def test_1_filesearch_default_parameters(self):
...
+                if 'File_1.txt' not in i:
```
Comment (kiranpawar7) [inline tempest/api/workloadmgr/snapshot/test_snapshot_mount.py]: This condition is not required, as disk is not mounted then no need to look for files.

File: tempest/api/workloadmgr/snapshot/test_snapshot_mount.py [hunk 6]
```diff
@@ -93,3 +93,137 @@ def test_1_filesearch_default_parameters(self):
...
+            LOG.error("Exception: " + str(e))
+            reporting.set_test_script_status(tvaultconf.FAIL)
+            reporting.test_case_to_write()
+
+    @test.attr(type='smoke')
+    @test.idempotent_id('215e0c36-8911-4167-aaea-8c07d21212f3')
+    def test_3_snapshot_mount_incremental(self):
+        reporting.add_test_script(str(__name__) + "_incremental_snasphot")
+        try:
+            if self.exception != "":
```
Comment (kiranpawar7) [inline tempest/api/workloadmgr/snapshot/test_snapshot_mount.py]: As same above this test case does not follow any prerequisite call, this condition can be removed.

File: tempest/api/workloadmgr/snapshot/test_snapshot_mount.py [hunk 7]
```diff
@@ -93,3 +93,137 @@ def test_1_filesearch_default_parameters(self):
...
+            instances_ids = self.instances_ids
+            snapshot_ids = self.snapshot_ids
+            wid = self.wid
+            volumes_ids = self.volumes_ids
+            security_group_id = self.security_group_id
+            fvm_id = self.fvm_id
+            incremental_snapshot_id = snapshot_ids[1]
+            floating_ips_list = self.floating_ips_list
+
+            LOG.debug("mount snasphot of a incremental snapshot")
```
Comment (kiranpawar7) [inline tempest/api/workloadmgr/snapshot/test_snapshot_mount.py]: String can be "Mount Incremental snapshot".

File: tempest/api/workloadmgr/snapshot/test_snapshot_mount.py [hunk 8]
```diff
@@ -93,3 +93,137 @@ def test_1_filesearch_default_parameters(self):
...
+            snapshot_ids = self.snapshot_ids
+            wid = self.wid
+            volumes_ids = self.volumes_ids
+            security_group_id = self.security_group_id
+            fvm_id = self.fvm_id
+            dismount_snapshot_id = snapshot_ids[0]
+            floating_ips_list = self.floating_ips_list
+
+            LOG.debug("dismount snasphot of a  snapshot")
+            is_dismounted = self.dismount_snapshot(dismount_snapshot_id)
```
Comment (savitha-p) [inline tempest/api/workloadmgr/snapshot/test_snapshot_mount.py]: There is no dismount_snapshot method in base.py. Please check.

Comment (maheshbhoge) [reply tempest/api/workloadmgr/snapshot/test_snapshot_mount.py]: commited function

**Pull request merged**